src/torrent.cpp

```diff
@@ -1675,54 +1558,28 @@ bool is_downloading_state(int const st)
 		// the verification function verifies the distinguished name
 		// of a peer certificate to make sure it matches the info-hash
 		// of the torrent, or that it's a "star-cert"
-		ctx->set_verify_callback(std::bind(&torrent::verify_peer_cert, this, _1, _2), ec);
```

Comment:
I think it would be nicer if the gnutls context would have `set_verify_callback()` and `verify_dn` would be a free function (or two free functions, but only one would be in use at a time).

I think it's an improvement to decouple it from being a `torrent` member function as you have done, but I think it should be decoupled from the context too.

Reply:
OK I'll see what I can do.